server/Controllers/GameControllers/snake.js

```diff
@@ -1,12 +1,117 @@
+const { SNAKE } = require('./gameConstants');
+
 class Snake {
-  constructor(playerId, initialPositionX, initialPositionY,
-    initialDirection, snakeColor, initialSnakeLength) {
+  /**
+   * @param  {string} playerId
+   * @param  {Object} initialProperties initial values for snake body and motion.
+   * @param  {Function} eatFood Function to check that snake's next movement will be eating food.
+   * @param  {Array} rivalBody Rival snake's body coordinates
```

Comment:
Avoid using comments. Comments are shown only when your code can not explain what it is doing. Try to use declarative variable names instead of comments. Use comments only when you are trying to do something tricky like rollback due to some weird reason or some code copy wasted from any source .

Reply:
Or are you using  for documentation like with JSDOC

Reply:
Yes, I am using JSDOC convention.

Reply:
Some variable names needed some explanation, so I thought I should give these variables some explanation.